server/log_operation_manager.go

```diff
@@ -8,87 +8,106 @@ import (
-// TODO(Martin2112): No mechanism for error reporting etc., this is OK for v1 but needs work
-func (l logOperationManager) OperationLoop() {
-	glog.Infof("Log operation manager starting")
+func NewLogOperationManager(done chan struct{}, sp LogStorageProviderFunc, batchSize int, sleepBetweenRuns time.Duration, timeSource util.TimeSource, logOperation LogOperation) *LogOperationManager {
+	return &LogOperationManager{context: LogOperationManagerContext{done: done, storageProvider: sp, batchSize: batchSize, sleepBetweenRuns: sleepBetweenRuns, timeSource: timeSource}, logOperation: logOperation}
+}
 
-	// Outer loop, runs until terminated
-	for {
-		// We might want to bail out early when testing
-		if l.runLimit >= 0 {
-			l.runLimit--
-			if l.runLimit < 0 {
-				return
-			}
-		}
+// For use by tests only, configures one shot mode
+func NewLogOperationManagerForTest(done chan struct{}, sp LogStorageProviderFunc, batchSize int, sleepBetweenRuns time.Duration, timeSource util.TimeSource, logOperation LogOperation) *LogOperationManager {
+	return &LogOperationManager{context: LogOperationManagerContext{done: done, storageProvider: sp, batchSize: batchSize, sleepBetweenRuns: sleepBetweenRuns, timeSource: timeSource, oneShot: true}, logOperation: logOperation}
+}
 
-		// Wait for the configured time before going for another pass
-		time.Sleep(l.sleepBetweenRuns)
+func (l LogOperationManager) getLogsAndExecutePass() bool {
+	// TODO(Martin2112) using log ID zero because we don't have an id for metadata ops
+	// this API could improved
+	provider, err := l.context.storageProvider(0)
 
-		// TODO(Martin2112) using log ID zero because we don't have an id for metadata ops
-		// this API could improved
-		provider, err := l.storageProvider(0)
+	// If we get an error, we can't do anything but wait until the next run through
+	if err != nil {
+		glog.Warningf("Failed to get storage provider for run: %v", err)
+		return false
+	}
 
-		// If we get an error, we can't do anything but wait until the next run through
-		if err != nil {
-			glog.Warningf("Failed to get storage provider for run: %v", err)
-			continue
-		}
+	tx, err := provider.Begin()
 
-		tx, err := provider.Begin()
+	if err != nil {
+		glog.Warningf("Failed to get tx for run: %v", err)
+		return false
+	}
 
-		if err != nil {
-			glog.Warningf("Failed to get tx for run: %v", err)
-			continue
-		}
+	// Inner loop is across all active logs, currently one at a time
+	logIDs, err := tx.GetActiveLogIDs()
 
-		// Inner loop is across all active logs, currently one at a time
-		logIDs, err := tx.GetActiveLogIDs()
+	if err != nil {
+		glog.Warningf("Failed to get log list for run: %v", err)
+		tx.Rollback()
+		return false
+	}
 
-		if err != nil {
-			glog.Warningf("Failed to get log list for run: %v", err)
-			tx.Rollback()
-			continue
-		}
+	if err := tx.Commit(); err != nil {
+		glog.Warningf("Failed to commit getting logs, continuing anyway: %v", err)
```

Comment:
This was inside a loop before it got refactored. Returning false does mean continue to run subsequent passes but it's not clear now so I've removed it.